numpy/_core/src/multiarray/item_selection.c

```diff
@@ -1137,7 +1116,7 @@ PyArray_Choose(PyArrayObject *ip, PyObject *op, PyArrayObject *out,
     PyDataMem_FREE(mps);
     if (out != NULL && out != obj) {
         Py_INCREF(out);
-        PyArray_ResolveWritebackIfCopy(obj);
+        PyArray_CopyAnyInto(out, obj);
```

Comment:
This can fail so you need to check the failure case. The same bug existed in the code you're updating - `ResolveWritebackIfCopy` also returns -1 on failure.

Reply:
I added a simple failure check; hopefully that is sufficient. Thank you!